spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -2751,8 +2753,8 @@ def keyPressEvent(self, event):
 
                     # Check if the line start with a comment or string
                     cursor = self.textCursor()
-                    cursor.setPosition(cursor.block().position(),

```

Comment:
Why is this code removed? That could have lead to unexpected results in other places.